### ISSUE TITLE
Add another thing

### ISSUE THREAD

**Issue by HelenOsg** [dwp/design-system-community-backlog#54]
## What
This pattern is trying to capture details for multiple items of the same things, for example the need to capture details of a user's job and then ask the user if they have another job -  building a picture of multiple jobs/employment.
It was found on Apply for New Style ESA replaying one of the data fields, the most relevant to the theme of the questions (in one example the employer name) helped the user identify which job they were giving information about.
Having a replay of the employer questions they have just answered "You've told us about XXXX" helped the user to separate individual job details
This pattern has been used in a number of question themes within the online form, anywhere where multiple items of a thing need to be captured E.G. Voluntary work, Pensions, Health Insurance.

### Example: Apply for New Style ESA
<img width="774" alt="Screen Shot 2021-03-22 at 09 29 14" src="https://user-images.githubusercontent.com/57486301/111968500-1e989680-8af1-11eb-9660-dbf5ad02a3e3.png">
<img width="781" alt="Screen Shot 2021-03-22 at 09 29 21" src="https://user-images.githubusercontent.com/57486301/111968506-20625a00-8af1-11eb-83a9-099703e59cd1.png">
<img width="755" alt="Screen Shot 2021-03-22 at 09 29 26" src="https://user-images.githubusercontent.com/57486301/111968514-22c4b400-8af1-11eb-978f-2b3279f55411.png">

### Example: NSJSA
<img width="721" alt="Screen Shot 2021-03-22 at 09 37 55" src="https://user-images.githubusercontent.com/57486301/111969584-518f5a00-8af2-11eb-9798-89a3009ac7dd.png">
<img width="650" alt="Screen Shot 2021-03-22 at 09 38 02" src="https://user-images.githubusercontent.com/57486301/111969593-53591d80-8af2-11eb-8cd1-854dc6afc0ad.png">

## Why

The  'Add to a list' pattern doesn't work for this as it's not adding one element but collecting a larger amount of data about users employment, such as; employer details, working pattern, any expenses paid and other data attributes. So we needed to have multiple data inputs associated with the job/s and this didn't fit with the existing 'Add another' pattern

## Anything else

**Original issue information:** 4 Feb 2019 @srich-DWP  https://github.com/dwp/design-system-community-backlog/issues/10

**Comment by htmlandbacon** [comment]
![working draft of add another](https://user-images.githubusercontent.com/4334015/133068174-1d125eb6-d57d-468a-9e05-777b7e45b577.png)

Following a workshop and review with @jon-houghton - we ended up with this.

**Comment by jonhurrell** [comment]
@abbott567 Do you have any suggestions or advice around how we can accessibly describe a multi-step user journey in an image, or is it an acceptable user experience to embedded (admittedly lengthy) content within an alt that describes the flows and UI over many screens? Is there a smarter way we can do this?

**Comment by abbott567** [comment]
## Detailed reply
@jonhurrell For any complex image it's generally accepted as best-practice to leave the alt attribute blank and describe the image in the body content. This way screen reader users can access structured content if you need to describe multiple steps with headings etc. By dumping it all in the alt attribute, structure like headings and lists etc is lost. 

It also means anybody else who doesn't understand the picture has access to the text alternative. By putting it in the alt attribute it isn't available to sighted users who might struggle to understand the image. GDS have recently changed their guidance to just do away with alt text in attributes all together. You can read the [updated GDS guidance on alt text](https://www.gov.uk/guidance/content-design/images#alt-text) in the content and publishing guidance on GOV.UK. 

It looks like the GDS Design System hasn't caught up yet and still has published content on alt text in the attribute, but I suspect that will change to align with the content design guidance at some point. I'll raise an issue to nudge them.

## TLDR (too long didn't read)
Give the image a blank alt attribute to mark it as decorative and hide it from screen readers. Describe everything the image represents in the body copy using properly structured content. Check out the [updated GDS guidance on alt text](https://www.gov.uk/guidance/content-design/images#alt-text).

**Comment by abbott567** [comment]
I've opened an [issue to align the guidance on alt text in the GOV.UK Design System](https://github.com/alphagov/govuk-design-system/issues/1887).

**Comment by jonhurrell** [comment]
@abbott567 Thanks, appreciate the feedback.

**Comment by jonhurrell** [comment]
[Published pattern](https://design-system.dwp.gov.uk/patterns/add-another-thing). You can contribute any research or insights using this [discussion](https://github.com/dwp/design-system-community-backlog/discussions/87)
